Labs6

New1sens opened the pull request.

New1sens committed:
Create Grafff.java

New1sens committed:
Delete Grafff.java

New1sens committed:
Save

integralpir commented on Второй семестр/labs6/Grafff.java:
```diff
@@ -0,0 +1,57 @@
+    }
+    //Добавляет ненаправленное ребро между двумя вершинами
+
+    public void addEdge(String vertex1, String vertex2) {
+        addVertex(vertex1);
+        addVertex(vertex2);
+
+        // Добавляем рёбра в обе стороны (ненаправленный граф)
+        adjacencyList.get(vertex1).add(vertex2);
+        adjacencyList.get(vertex2).add(vertex1);
+    }
```
Если ребра не существует вылетит NPE, нужно исправить

New1sens committed:
Update Grafff.java

Теперь граф устойчив к NPE и автоматически создаёт вершины при добавлении рёбер.

integralpir commented on Второй семестр/labs6/Grafff.java:
```diff
@@ -0,0 +1,66 @@
+        Objects.requireNonNull(vertex1, "vertex1 не может быть null");
+        Objects.requireNonNull(vertex2, "vertex2 не может быть null");
+
+        addVertex(vertex1);
+        addVertex(vertex2);
```
ребра обычно добавляют для существующих вершин, надо добавить проверку на существование переданной вершины.

integralpir commented on Второй семестр/labs6/Grafff.java:
```diff
@@ -0,0 +1,66 @@
+        Objects.requireNonNull(vertex1, "vertex1 не может быть null");
+        Objects.requireNonNull(vertex2, "vertex2 не может быть null");
+
+        addVertex(vertex1);
+        addVertex(vertex2);
```
тут тоже

New1sens committed:
Update Grafff.java

New1sens committed:
Delete Второй семестр/labs6/Readme.md

integralpir commented on Второй семестр/labs6/Grafff.java:
```diff
@@ -0,0 +1,70 @@
+package labs6;
+import java.util.*;
+
+class Graph {
```
Зачем модификатор доступа убрал?

integralpir reviewed (approved):
Ладно, зачту, но исключение если вершина не существует это лишнее, можно было просто создавать ее в методе с добавлением ребра
